functional.py

```diff
@@ -0,0 +1,62 @@
+import sys
+
+def sequential_map(*args):
+    object = args[-1]
+    if type(object) == int:
+        for func in args[:-1]:
+            object = func(object)
+    else:
+        for func in args[:-1]:
+            object = list(map(func, object))
+    return object
+
+
+def func_chain(*args):
+    def result(object):
+        if type(object)==int:  #for numbers
+            for func in args:
+                object = func(object)
+        else:  #for lists and tuples
+            for func in args:
+                object = list(map(func, object))
+        return object
+    return result
+
+
+def consensus_filter(*args):
+    object = args[-1]
+    if type(object) == int:  # for integer objects
+        if args[0](object):
+            return object
+        else:
+            return None
+    for func in args[:-1]:  # for sets and lists
+        right_nums = list(map(func, object))  # list of True and False
+        indexes = [i for i, x in enumerate(right_nums) if x]  # indexes of right elems in object
+        object = [object[ind] for ind in indexes]  # create new, reduced object
+    return object
+
+
+def conditional_reduce(func1, func2, object):
+    right_nums = list(map(func1, object))  # list of True and False
+    indexes = [i for i, x in enumerate(right_nums) if x]  # indexes of right elems in object
+    object_for_func2 = [object[ind] for ind in indexes]  # create new, reduced object
```

Comment:
Какой-то странный велосипед, у нас же есть функция filter
```suggestion
    object_for_func2 = filter(func1, object) 
```